src/js/components/AppPageComponent.jsx

```diff
@@ -31,17 +31,29 @@ var statusNameMapping = {
 var AppPageComponent = React.createClass({
   displayName: "AppPageComponent",
 
-  propTypes: {
-    appId: React.PropTypes.string.isRequired,
-    router: React.PropTypes.object.isRequired,
-    view: React.PropTypes.string
+  contextTypes: {
+    router: React.PropTypes.func
   },
 
   getInitialState: function () {
-    var activeTabId;
-    var appId = this.props.appId;
+    return _.extend({
+      fetchState: States.STATE_LOADING
+    }, this.getRouteSettings(this.props));
```

Comment:
`fetchState` will never be returned from `#getRouteSettings` therefore this `extend` is just doing unnecessary work.
We can just do:

```
var settings = this.getRouterSettings(this.props);
settings.fetchState = States.STATE_LOADING;
return settings;
```